feature/session/src/main/res/layout/fragment_bottom_sheet_sessions.xml

```diff
@@ -11,78 +11,125 @@
-            app:layout_constraintBaseline_toBaselineOf="@id/sessions_bottom_sheet_title"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:textEndPadding="@dimen/session_bottom_sheet_filter_text_padding"
-            app:textStartPadding="@dimen/session_bottom_sheet_filter_text_padding"
-            />
+                android:id="@+id/sessions_bottom_sheet_hide_filter_button"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginEnd="16dp"
+                android:gravity="center_vertical"
+                android:paddingEnd="@dimen/session_bottom_sheet_filter_text_padding"
+                android:paddingStart="@dimen/session_bottom_sheet_filter_text_padding"
+                app:visibleGone="@{isCollapsed &amp;&amp; !shouldShowEmptyStateView}"
+                app:chipBackgroundColor="@color/gray4"
+                app:chipIcon="@drawable/ic_arrow_upward_black_24dp"
+                app:chipIconTint="@color/gray3"
+                app:iconEndPadding="@dimen/session_bottom_sheet_filter_icon_padding"
+                app:iconStartPadding="@dimen/session_bottom_sheet_filter_icon_padding"
+                app:layout_constraintBaseline_toBaselineOf="@id/sessions_bottom_sheet_title"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:textEndPadding="@dimen/session_bottom_sheet_filter_text_padding"
+                app:textStartPadding="@dimen/session_bottom_sheet_filter_text_padding"
+        />
 
         <androidx.recyclerview.widget.RecyclerView
-            android:id="@+id/sessions_recycler"
-            android:orientation="vertical"
-            android:layout_width="0dp"
-            android:layout_height="0dp"
-            android:layout_marginTop="16dp"
-            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/sessions_bottom_sheet_title"
-            />
+                android:id="@+id/sessions_recycler"
+                android:orientation="vertical"
+                android:layout_width="0dp"
+                android:layout_height="0dp"
+                android:layout_marginTop="16dp"
+                app:visibleGone="@{!shouldShowEmptyStateView}"
+                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/sessions_bottom_sheet_title"
+        />
+
+        <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Nested ConstraintLayout might be confusing.
It would be enough to use a vertical LinearLayout whose height is wrap_content. 
`app:layout_constraintVertical_bias="0.42"` should be attached to this LinearLayout then.

Reply:
Actually LinearLayout is better and enough for this situation. Fixed at https://github.com/DroidKaigi/conference-app-2019/pull/309/commits/29a5a4508a6f943b8033116b5a9279e1af429c2c .